kotlin/edit-with-branch-versioning/src/main/java/com/esri/arcgisruntime/sample/editwithbranchversioning/MainActivity.kt

```diff
@@ -0,0 +1,354 @@
+/*
+ * Copyright 2020 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgisruntime.sample.editwithbranchversioning
+
+import android.content.DialogInterface
+import android.os.Bundle
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.MotionEvent
+import android.view.View
+import android.widget.ArrayAdapter
+import android.widget.EditText
+import android.widget.Spinner
+import android.widget.Toast
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.app.AppCompatActivity
+import com.esri.arcgisruntime.arcgisservices.ServiceVersionParameters
+import com.esri.arcgisruntime.arcgisservices.VersionAccess
+import com.esri.arcgisruntime.data.Feature
+import com.esri.arcgisruntime.data.ServiceGeodatabase
+import com.esri.arcgisruntime.layers.FeatureLayer
+import com.esri.arcgisruntime.loadable.LoadStatus
+import com.esri.arcgisruntime.mapping.ArcGISMap
+import com.esri.arcgisruntime.mapping.Basemap
+import com.esri.arcgisruntime.mapping.Viewpoint
+import com.esri.arcgisruntime.mapping.view.DefaultMapViewOnTouchListener
+import com.esri.arcgisruntime.security.AuthenticationManager
+import com.esri.arcgisruntime.security.DefaultAuthenticationChallengeHandler
+import kotlinx.android.synthetic.main.activity_main.*
+
+class MainActivity : AppCompatActivity() {
+
+  private val TAG = MainActivity::class.java.simpleName
+
+  private val serviceGeodatabase: ServiceGeodatabase by lazy {
+    ServiceGeodatabase("https://sampleserver7.arcgisonline.com/arcgis/rest/services/DamageAssessment/FeatureServer").apply {
+      loadAsync()
+    }
+  }
+
+  private var featureLayer: FeatureLayer? = null
+  private var selectedFeature: Feature? = null
+  private var createdVersionName: String = ""
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_main)
+
+    // create a default authentication challenge handler because this service requires authentication
+    AuthenticationManager.setAuthenticationChallengeHandler(
+      DefaultAuthenticationChallengeHandler(
+        this
+      )
+    )
+
+    // create a map with a streets vector basemap and set it to the mapview
+    mapView.map = ArcGISMap(Basemap.createStreetsVector())
```

Comment:
I'd suggest moving this down to an apply block with the other mapView/mapView.map work at lines 91 and 97